### ISSUE TITLE
INVALID_LOGIN

### ISSUE THREAD

**Issue by InfoSecured** [simple-salesforce/simple-salesforce#171]
Hi,

I'm trying to login use simple-salesforce and no matter what I do, I get this error:
`simple_salesforce.login.SalesforceAuthenticationFailed: INVALID_LOGIN: Invalid username, password, security token; or user locked out.`

I've tried python 2.7 and 3.6.  I'm passing in this code:
`import pandas as pd
import numpy as np
import sys
#import salesforce as sf
#import config
#sys.path.insert=('salesforce/')

from simple_salesforce import Salesforce

sf=Salesforce(username='my username', password='my password', security_token='my token')

print('login successful')`

I saw in someone's thread that they had an openssl issue that caused this but mine is up to date.  I've even tried on different computers.

Any thoughts?

**Comment by afs2015** [comment]
Have you tried just running python from the command line and done the following with manually typing in your security credentials versus importing them from config?

from simple_salesforce import Salesforce
sf=Salesforce(username='my username', password='my password', security_token='my token')

**Comment by InfoSecured** [comment]
Yes, I get the same error.

**Comment by skamensky** [comment]
Try logging in from the browser on the same computer. Do you have IP filtering on?

**Comment by InfoSecured** [comment]
I can log in with my browser and we do use IP filtering, however this IP is whitelisted.

**Comment by skamensky** [comment]
If you're using ip filtering you need to include `organizationId`in your initialization parameters in order for simple_python to properly construct the SOAP login message which has a special `LoginScopeHeader `attribute.

So something like

`sf=Salesforce(username='myusername@something.com', password='mypassword', security_token='my token',organizationId='myorgId')`

Let me know if that works!

**Comment by InfoSecured** [comment]
Still no good. Same error.

**Comment by InfoSecured** [comment]
Is there anything I need to pass for a custom domain?

**Comment by skamensky** [comment]
No, you shouldn't need to add any special information for custom domain. login.py will handle that.

Are you logging into a sandbox? If so, do you have `sandbox `set to true?
Are you entering your password as you regularly would when logging in? Sometimes people get confused and add the `security_token `at the end of their password. But that's not necessary since login.py handles that for you.

**Comment by InfoSecured** [comment]
I'm not logging into a sandbox.  I'm putting my password in separate from the security_token.

**Comment by skamensky** [comment]
Is your ip address whitelisted by profile or by network access?

If it's filtered by profile, your security token will be invalidated. You can check by confirming that the 

_reset my security token_ option exists under _my settings > personal_ in salesforce

Another thing you can do to make sure your security token is working properly is to try and log in with another application that requires a security token (like force.com IDE or Apex data loader)

**Comment by InfoSecured** [comment]
So, unfortunately I don't have access to force.com or the Apex data loader.  I've been able to try this from both a whitelisted IP and non-whitelisted IP and am getting the same error.  I've reset my security token twice now just to make sure it wasn't a bad token.  Do I need specific access permissions in salesforce to actually make this work?

**Comment by paulmorriss** [comment]
I'm getting the same problem. My IP address is whitelisted by profile, so I have no token, and I can't log in specifying the organizationId. I see in the code for login.py that you can login without an org id, but the code in api.py doesn't let you get that far if you don't specify an org id.

**Comment by xenon234** [comment]
Not sure if this relates to OP's problem, but I was having trouble with token-less login until i added the line `security_token=''` (those are two single quotes, not one double quote) to the `Salesforce()` constructor call

**Comment by clivebearmanhub** [comment]
@gredondogc How'd ya get it to work?  I've been trying for hours, but with zero success.  Thanks

**Comment by noah-seltzer** [comment]
@clivebearmanhub simplesalesforce() constructor throws an error if there's no security_token parameter. There are currently pull requests open to re-design the function but for now just adding security_token='' inside the brackets is a workaround for now

**Comment by clivebearmanhub** [comment]
@noah-seltzer Thanks.  Works now.

**Comment by phpepe** [comment]
In my case just adding domain='test' did the trick (for sandbox)

**Comment by sudevpdas** [comment]
> In my case just adding domain='test' did the trick (for sandbox)

Yes, it works.

**Comment by vlookup** [comment]
> In my case just adding domain='test' did the trick (for sandbox)

The year is 2023 and there is a pathetic soul struggling with all his might, until he runs across this simple but wonderful nugget that allows him to print the session id and instance to validate connection.

**Comment by RapidFire05** [comment]
THis post helped me: https://stackoverflow.com/a/60460085/2252729
```
from simple_salesforce import Salesforce

USER = "user@domain.com.sandbox_name"
PASS = "pass"
SEC_TOKEN = "token"
DOMAIN = "<domain>--<sandbox_name>.<instance>.my"

sf = Salesforce(username=USER, password=PASS, security_token=SEC_TOKEN, domain=DOMAIN)
```

**Comment by kleicht** [comment]
adding "my" at the end worked for me, thx !
